terraform/provider/provider.go

```diff
@@ -37,42 +42,100 @@ func Provider() *schema.Provider {
 			},
 		},
 		ResourcesMap: map[string]*schema.Resource{
-			"teleport_user": resourceTeleportUser(),
 			"teleport_role": resourceTeleportRole(),
+			"teleport_user": resourceTeleportUser(),
 		},
-		ConfigureFunc: providerConfigure,
+		ConfigureContextFunc: configureContext,
 	}
 }
 
-// providerConfigure accepts provider settings schema,
+// configureContext accepts provider settings schema,
 // initializes a Teleport's `auth.Client` and returns it.
-func providerConfigure(d *schema.ResourceData) (interface{}, error) {
-	certPath := d.Get("cert_path").(string)
-	keyPath := d.Get("key_path").(string)
-	rootCAsPath := d.Get("root_ca_path").(string)
+func configureContext(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	certPath, err := getPath(d, "cert_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
+
+	keyPath, err := getPath(d, "key_path")
```

Comment:
Nit: I'd consider moving `cert_path`, `key_path`, `root_ca_path` and `addr` keys into constants.

Reply:
Makes sense, moved